wildstore-cli/src/test/java/edu/sjsu/wildstore/CliTest.java

```diff
@@ -237,6 +246,128 @@ void testShare() throws IOException {
         deleteUsers();
     }
 
+    @Test
+    void testClean() throws IOException, URISyntaxException, ExecutionException, InterruptedException {
+        var cmd = new Main.Cli();
+        var adminTokenFile = tempDir.resolve("admin-token.txt");
+        var userTokenFile = tempDir.resolve("user-token.txt");
+
+        var adminToken = "secret-admin-token";
+        var userToken = "secret-user-token";
+
+        Files.write(adminTokenFile, ("token=" + adminToken).getBytes());
+        Files.write(userTokenFile, ("token=" + userToken).getBytes());
+
+        createUser("ROLE_ADMIN", "CleanAdmin", "admin@clean", adminToken);
+        createUser("ROLE_USER", "ClearUser", "user@clean", userToken);
+
+        //put files into directory
+        var testDataUtils = new TestDataUtils();
+        testDataUtils.extractTestData(tempDir);
+
+        //find all .nc files in the tempDir
+        List<String> fileNames = List.of();
+        try (Stream<Path> walk = Files.walk(tempDir)) {
+            fileNames = walk.filter(path -> path.toString().endsWith(".nc")).map(Path::toString).toList();
+        }
+        if (fileNames.isEmpty()) {
+            throw new IOException("No .nc files found in the test data directory.");
+        }
+
+        //creates a file with the filenames to crawl
+        var nameFile = tempNameDir.resolve("fileNames.txt");
+        Files.write(nameFile, String.join("\n", fileNames).getBytes());
+
+        //crawl files
+        try {
+            for (int i = 0; i < Math.min(2, fileNames.size()); i++) {
+                WildfireFilesCrawler.crawl(fileNames.get(i), Client.getWebClient(metaURL + "/api/metadata"),
+                                           userToken, 1024 * 1024, "all", false);
+            }
+        } catch (ExecutionException e) {
+            throw new RuntimeException(e);
```

Comment:
rethrowing a new RuntimeException is code smell. sometimes you have to, but in this case, you can add the thrown exceptions to testClean() you shouldn't need to throw any new RuntimeExceptions in this code.